src/lib/documents.ts

```diff
@@ -72,3 +112,64 @@ export function isNIT(str: string): boolean {
 
   return Number(verifier) === sum;
 }
+
+/**
+ * Verifies that given NIT with hyphens is valid
+ *
+ * @param  {string} str       A string representing NIT digits (with hyphens)
+ * @returns {boolean}         Validity of the given NIT
+ */
+function isNITWithHyphen(str: string): boolean {
+  if (!nitRegExp.test(str)) return false;
+  const [municipality, birthdate, correlative, verifier] = str.split('-');
+  return isValidNIT(municipality, birthdate, correlative, verifier);
+}
+
+/**
+ * Verifies that given NIT in numeric format is valid
+ *
+ * @param  {string} str       A string representing NIT digits (without hyphens)
+ * @returns {boolean}         Validity of the given NIT
+ */
+function isNITWithNumbers(str: string): boolean {
+  if (!(str.length === 14)) return false;
+  const municipality = str.slice(0, 4);
+  const birthdate = str.slice(4, 10);
+  const correlative = str.slice(10, 13);
+  const verifier = str.slice(13);
+  return isValidNIT(municipality, birthdate, correlative, verifier);
+}
+
+/**
+ * Verifies that given NIT format is valid
+ *
+ * @param  {string} str       A string representing NIT digits (with hyphen)
+ * @returns {boolean}         Validity of the given NIT
+ */
+export function isNIT(str: string): boolean {
+  if (Number(str)) return isNITWithNumbers(str);
```

Comment:
As I mentioned before this feature should be submitted as a separate PR but I would recommend using RegExp instead of relying on `falsy` values (`0`, `NaN`, etc)